### PR TITLE
[JENKINS-58732] Removing Docker agent functionality

### PR DESCRIPTION
**Pull request by jglick**
[JENKINS-58732](https://issues.jenkins-ci.org/browse/JENKINS-58732)

Downstream of https://github.com/jenkinsci/docker-workflow-plugin/pull/202 for logistical reasons; see that PR’s description for a release guide.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -141,7 +141,7 @@
       <dependency>
         <groupId>org.jenkins-ci.plugins</groupId>
         <artifactId>docker-workflow</artifactId>
-        <version>1.14</version>
+        <version>1.22-rc452.c6ab7e23b1c4</version> <!-- TODO https://github.com/jenkinsci/docker-workflow-plugin/pull/202 -->
```
Comment (jglick) [inline pom.xml]: Must be released first!

**Comment by jglick** [comment]
`ProvisioningTimeoutModelDefTest` added recently in #376, failures probably unrelated.

File: pom.xml (hunk 1 above)
Comment (dwnusbaum) [reply pom.xml]: 1.22 was just released:

```suggestion
        <version>1.22</version>
```

**Comment by jglick** [comment]
Timeouts look like flakes. I suspect `ProvisioningTimeoutModelDefTest` is still a contributing factor after #379 but it is hard to say.

**Comment by joseblas** [comment]
Will be removed shortly

**Comment by dwnusbaum** [comment]
Rebuilding now that #381 was merged.

**Comment by jglick** [comment]
Seems to have worked this time.

**Pull request merged**